### ISSUE TITLE
Implement complex items interaction (cards stack)

### ISSUE THREAD

**Issue by bd-arc** [archriss/react-native-snap-carousel#195]
The goal is to implement the following kind of interaction, relying only on the `FlatList` component and on `Animated` interpolations. It might not be possible, but we need to find out :-)

[Event-cards-iOS-interaction, by Divan Raj](https://dribbble.com/shots/3731362-Event-cards-iOS-interaction)
![react-native-snap-carousel advanced layout idea](https://i.imgur.com/rcS3VAzl.png)

[Onboarding, by Luobing](https://dribbble.com/shots/3891675-Onboarding)
![react-native-snap-carousel advanced layout idea](https://i.imgur.com/xEFTmlKl.gif)

[InVision Studio - Movies app concept, by Charles Patterson for InVision](https://dribbble.com/shots/3982621-InVision-Studio-Movies-app-concept)
![react-native-snap-carousel advanced layout idea](https://i.imgur.com/jA5jg1Zl.png)

**Comment by bd-arc** [comment]
See also #244

**Comment by bd-arc** [comment]
And yes, this is possible 🎉 

![react-native-snap-carousel custom animations](https://i.imgur.com/Ta3xHCn.gif)

```
    _initPositionsAndInterpolators (props = this.props) {

        [...]

            } else {
                animatedValue = this._scrollPos.interpolate({
                    inputRange: [
                        (_index - 3) * sizeRef,
                        (_index - 2) * sizeRef,
                        (_index - 1) * sizeRef,
                        _index * sizeRef,
                        (_index + 1) * sizeRef
                    ],
                    outputRange: [3, 2, 1, 0, -1],
                    extrapolate: 'clamp'
                });
            }

            interpolators.push(animatedValue);
        });

        this.setState({ interpolators });
    }
```

```
    _renderItem ({ item, index }) {
        const { interpolators } = this.state;
        const {
            data,
            ...
        } = this.props;

        const animatedValue = interpolators && interpolators[index];

        if (!animatedValue && animatedValue !== 0) {
            return false;
        }

        [...]

        const animatedStyle = animate ? {
            zIndex: data.length - index,
            opacity: animatedValue.interpolate({
                inputRange: [-1, 0, 1, 2, 3],
                outputRange: [1, 1, 0.75, 0.5, 0]
            }),
            transform: [{
                scale: animatedValue.interpolate({
                    inputRange: [-1, 0, 1, 2, 3],
                    outputRange: [0.9, 1, 0.9, 0.8, 0.8]
                })
            }, {
                translateX: animatedValue.interpolate({
                    inputRange: [-1, 0, 1, 2, 3],
                    outputRange: [
                        -itemWidth * 0.5,
                        0,
                        -itemWidth * 0.99,
                        -itemWidth * 2.25,
                        -itemWidth * 3.45
                    ]
                })
            }]
        } : {};

        [...]
    }
```

I'm now going to work on adding the ability to use a custom interpolator like this one.

**Comment by bd-arc** [comment]
Feature released in [version `3.6.0`](https://github.com/archriss/react-native-snap-carousel/releases/tag/v3.6.0).
